profiling/src/profile/internal/observation/observations.rs

```diff
@@ -118,9 +131,9 @@ mod test {
         o.add(s2, None, vec![7, 8, 9]);
         o.iter().for_each(|(k, ts, v)| {
             assert!(ts.is_none());
-            if k == s1 {
+            if *k == s1 {
```

Comment:
Notably, if you returned `Sample` intead of `&Sample`, you wouldn't have to do `*k` :)

Reply:
done